### PR TITLE
Creating streamlet secrets if missing in cli

### PR DESCRIPTION
**Pull request by RayRoestenburg**
### What changes were proposed in this pull request?

Create empty secrets before creating input secret.
cc @andreaTP need to do this as well in the new cli.

### Why are the changes needed?

This solves an ordering issue with the operator needing to get the secret before creating runner resource (deployment, flinkapp, sparkapp)

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

- [x] GKE

### CODE REVIEW

File: kubectl-cloudflow/cmd/deploy_command.go [hunk 1]
```diff
@@ -194,7 +194,7 @@ func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	ownerReference := createOrUpdateCloudflowApplication(appClient, applicationSpec)
 
 	appInputSecret = config.UpdateSecretWithOwnerReference(ownerReference, appInputSecret)
-
+	createStreamletSecretsIfMissing(k8sClient, applicationSpec, ownerReference)
```
Comment (andreaTP) [inline kubectl-cloudflow/cmd/deploy_command.go:197]: adding this method in this place makes the method below to always be an update, correct?

File: kubectl-cloudflow/config/config.go [hunk 2]
```diff
@@ -694,13 +694,40 @@ func addCommandLineArguments(spec cfapp.CloudflowApplicationSpec, config *Config
...
+		Type: corev1.SecretTypeOpaque,
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: appID,
+			Labels:    labels,
+		},
+	}
+	secretMap := make(map[string]string)
+	secretMap["secret.conf"] = "{}"
+	secretMap["runtime-config.conf"] = "{}"
```
Comment (andreaTP) [inline kubectl-cloudflow/config/config.go:722]: If I read this change correctly, this is the core part of it, have you tested if just adding those defaults to `createOrUpdateAppInputSecret` make it?

File: kubectl-cloudflow/cmd/deploy_command.go (hunk 1 above)
Comment (RayRoestenburg) [reply kubectl-cloudflow/cmd/deploy_command.go:197]: No that is a different secret. (input for all streamlets vs secret for streamlet). This has to be improved further TBH.

File: kubectl-cloudflow/config/config.go (hunk 2 above)
Comment (RayRoestenburg) [reply kubectl-cloudflow/config/config.go:722]: n streamlets -> n empty secrets. 1 app secret.

**Pull request merged**